posthog/models/feature_flag/flag_status.py

```diff
@@ -0,0 +1,175 @@
+        self.feature_flag_id = feature_flag_id
+        self.stale_window = stale_window
+
+    def get_status(self) -> tuple[FeatureFlagStatus, FeatureFlagStatusReason]:
+        try:
+            flag = FeatureFlag.objects.get(pk=self.feature_flag_id)
+        except FeatureFlag.DoesNotExist:
+            return FeatureFlagStatus.UNKNOWN, "Flag could not be found"
+
+        if flag.deleted:
+            return FeatureFlagStatus.DELETED, "Flag has been deleted"
+
+        # See if the flag is set to 100% on one variant (or 100% on rolled out and active if boolean flag).
+        is_flag_fully_enabled, fully_enabled_explanation = self.is_flag_fully_enabled(flag)
+        if is_flag_fully_enabled:
+            return FeatureFlagStatus.STALE, fully_enabled_explanation
+
+        # Final, and most expensive check: see if the flag has been evaluated recently.
+        if self.is_flag_unevaluated_recently(flag):
+            return FeatureFlagStatus.INACTIVE, "Flag has not been evaluated recently"
+
+        return FeatureFlagStatus.ACTIVE, "Flag is not fully rolled out and may still be actively called"
+
+    def is_flag_fully_enabled(self, flag: FeatureFlag) -> tuple[bool, FeatureFlagStatusReason]:
+        # If flag is not active, it is not enabled. This flag may still be stale,
+        # but only if isn't being evaluated, which will be determined later.
+        if not flag.active:
+            logger.debug(f"Flag {flag.id} is not active")
+            return False, ""
+
+        # If flag is using super groups and any super group is rolled out to 100%,
+        # it is fully enabled.
+        if flag.filters.get("super_groups", None):
+            for super_group in flag.filters.get("super_groups"):
+                if self.is_group_fully_rolled_out(super_group):
+                    logger.debug(f"Flag {flag.id} has super group is rolled out to 100%")
+                    return True, "Super group is rolled out to 100%"
+
+        # If flag is using holdout groups and any holdout group is rolled out to 100%,
+        # it is fully enabled.
+        if flag.filters.get("holdout_groups", None):
+            for holdout_group in flag.filters.get("holdout_groups"):
+                if self.is_group_fully_rolled_out(holdout_group):
+                    logger.debug(f"Flag {flag.id} has holdout group is rolled out to 100%")
+                    return True, "Holdout group is rolled out to 100%"
+
+        multivariate = flag.filters.get("multivariate", None)
+        if multivariate:
+            is_multivariate_flag_fully_enabled, fully_enabled_variant_name = self.is_multivariate_flag_fully_enabled(
+                flag
+            )
+        if multivariate and is_multivariate_flag_fully_enabled:
+            return True, f'Currently, this flag will always use the variant "{fully_enabled_variant_name}"'
+        elif self.is_boolean_flag_fully_enabled(flag):
+            return True, 'Currently, this boolean flag will always evaluate to "true"'
```

Comment:
do we need "Currently"?  Feels like that's a superfluous word.

Reply:
Yep fair